test/index.js

```diff
@@ -1117,6 +1118,60 @@ describe('list', function() {
 
   });
 
+  describe('slice', function() {
+
+    it('returns a Nothing if applied to an empty list', function() {
+      assert(S.slice(0, 3, []).equals(S.Nothing()));
+    });
```

Comment:
This is misleading since `S.slice(0, 0, [])` evaluates to `Just([])`. I think we could remove this assertion.